config/config.go

```diff
@@ -170,7 +171,7 @@ func evaluateConfigWithEnv(configFile io.Reader, writers ...io.Writer) (io.Reade
 		return nil, fmt.Errorf("unable to read the config file: %w", err)
 	}
 
-	t := template.New("appConfigTemplate")
+	t := template.New("appConfigTemplate").Option("missingkey=zero")
```

Comment:
This is neat, previously the template was setting the value to `<no value>` when the env var the template needed wasn't set, now it defaults to the "zero value" for whatever type (so we actually get an empty string that can be caught by the `validate:"required"` tag)